app/controllers/api/v2/CorsProxyController.ts

```diff
@@ -0,0 +1,64 @@
+import { Readable } from 'stream';
+import { ReadableStream } from 'stream/web';
+
+import { Context } from 'koa';
+import { Duration } from 'luxon';
+
+import { BadRequestException, ValidationException } from '../../../support/exceptions';
+import { currentConfig } from '../../../support/app-async-context';
+
+const headersToPass = ['Location', 'Content-Type', 'Content-Length'];
+const fallbackTimeoutMs = 1000;
+
+export async function proxy(ctx: Context) {
+  const {
+    timeout: timeoutString,
+    allowedOrigins,
+    allowedURlPrefixes,
+    allowLocalhostOrigins,
+  } = currentConfig().corsProxy;
+
+  const { origin } = ctx.headers;
+
+  if (typeof origin !== 'string') {
+    if (!allowedOrigins.includes('none')) {
+      throw new BadRequestException('Missing origin');
```

Comment:
Status 400 means that user has a chance to fix the situation. But it's not the case as origin is not manually chosen. 

We need some other Status I believe

Reply:
I don't think that we have any better option. It is the client error (definitely not the server one), i.e. some of the 4xx codes. The 400 means 'some problem with the request that we can't specify more clearly by other 4xx codes'. 

Of the other codes, only the 403 is more or less fits.